mifosng-provider/src/main/java/org/mifosplatform/infrastructure/documentmanagement/service/ImageReadPlatformServiceImpl.java

```diff
@@ -45,9 +49,14 @@ public ImageMapper(final String entityDisplayName) {
             this.entityDisplayName = entityDisplayName;
         }
 
-        public String schema() {
-            return " image.id as id, image.location as location, image.storage_type_enum as storageType "
-                    + " from m_image image , m_client client " + " where client.image_id = image.id and client.id=?";
+        public String schema(String clientName) {
+            StringBuilder builder = new StringBuilder("image.id as id, image.location as location, image.storage_type_enum as storageType ");
+            if (ImageWritePlatformServiceJpaRepositoryImpl.IMAGE_TYPE.CLIENTS.toString().equals(clientName)) {
+                builder.append(" from m_image image , m_client client " + " where client.image_id = image.id and client.id=?");
+            } else if (ImageWritePlatformServiceJpaRepositoryImpl.IMAGE_TYPE.STAFF.toString().equals(clientName)) {
+                builder.append("from m_image image , m_staff staff " + " where staff.image_id = image.id and staff.id=?");
+            }
```

Comment:
Not a good idea to discriminate with conditionals like this (will have a bad impact down the road). Have a look at: http://www.refactoring.com/catalog/replaceConditionalWithPolymorphism.html